File: lose/compiler.py
from antlr4 import *
import sys
import os
from assignxgen import writeAssignsUpToX
from LoseLexer import LoseLexer
from LoseParser import LoseParser
from LoseListener import LoseListener
from codewriter import *
import time

def main(argv):
    input = FileStream(argv[1])
    lexer = LoseLexer(input)
    stream = CommonTokenStream(lexer)
    parser = LoseParser(stream)
    tree = parser.prog()

    writer = CodeWriter()
    walker = ParseTreeWalker()
    walker.walk(writer, tree)

if __name__ == '__main__':
    main(sys.argv)

File: lose/Compiler.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.*;

class Compiler {
    public static void main(String[] args) {
        LoseLexer lexer = null;
        try {
            lexer = new LoseLexer(new ANTLRFileStream(args[0]));
        } catch (IOException e) {
            e.printStackTrace();
        }

        LoseParser parser = new LoseParser(new CommonTokenStream(lexer));
        ParseTree tree = parser.prog();
        ParseTreeWalker.DEFAULT.walk(new CodeWriter("",""), tree);
    }
}
